src/Registration/pReg/Reg.py.in

```diff
@@ -237,6 +237,11 @@ class NiftiImageData(SIRF.ImageData):
         """Get sum."""
         return parms.float_par(self.handle, 'NiftiImageData', 'sum')
 
+    def dimensions(self):
+        """Returns image dimensions as a tuple (nz, ny, nx)."""
```

Comment:
I dislike `nz,ny.nx` terminology. What do x,y,z mean anyway?

Is it equivalent to `shape()`? (but faster). I hope so. I would state that in the help. If it is equivalent, then `shape` should alias to this one I guess

See also

Reply:
@KrisThielemans  I copy pasted it from other `dimensions()` definitions. I think it is equivalent, not sure in other cases. This one does not in fact return `(nz,ny,nz)` but behaves like shape, returning only if dimensions are bigger than `1` (i.e. a 2D array would output a 2 element shape in Reg.py, but I am not sure in the others). 

But mentioning it in the help is fair. I suggest that instead of doing that here, we open a new issue. There are several instances of the `dimensions()` function, so that way we can tackle all of them at the same time. 

I am not sure if I can alias `shape`, as it is a `@property` of `DataConainer` while `dimensions()` is only defined in each class that inherits from `DataConainer`. In any case, same comment as before, there are several`dimensions()` so maybe better in a new issue/PR?

Reply:
I created [`shape`](https://github.com/SyneRBI/SIRF/blob/master/src/common/SIRF.py#L508) for compatibility with CIL and it goes through numpy.

Reply:
`shape` is definitely a natural for numpy. The relations need to be made clear of course. I think it'd be great to least somewhere which are the Python-specific additions made for numpy/CIL compatibility (but I also think this is already done in the `UsersGuide.md`.

It seems that `dimensions` needs to exist at the top level (a `virtual` in C++).

sure, a separate PR is fine. refer to this discussion by using a link.

Reply:
can't `shape` use `get_dimensions`? Mine was just a quick trick.

Reply:
@paskino `shape` sits in `DataConainer` as a `@property` and `get_dimensions` and `dimensions` in the object that inherits from `DataConainer`. Can you make `shape` use those methods?